### PR TITLE
优化主机属性自动应用预览接口

### PR DESCRIPTION
**Pull request by laodiu**
### 新加的功能:
- xxxx
### 优化的功能:
- 主机应用预览接口性能优化
### 修复的问题：
- xxxx

### 优化方式
- 通过减少查询主机字段，减少了db查询时间，减少了数据包的大小，减少了数据包的传输时间。

close issue  #5404

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/bk-cmdb?pullRequest=5403) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/bk-cmdb?pullRequest=5403) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/bk-cmdb?pullRequest=5403) it.</sub>

**Comment by laodiu** [comment]
优化主机属性自动应用预览接口

File: src/readme.md [hunk 1]
```diff
@@ -0,0 +1 @@
+测试gpg签名
```
Comment (fireyun) [inline src/readme.md:1]: 去掉该文件

File: src/source_controller/coreservice/core/hostapplyrule/plan.go [hunk 2]
```diff
@@ -48,8 +48,27 @@ func (p *hostApplyRule) GenerateApplyPlan(kit *rest.Kit, bizID int64, option met
 			common.BKDBIN: hostIDs,
 		},
 	}
+	// get attributes
```
Comment (fireyun) [inline src/source_controller/coreservice/core/hostapplyrule/plan.go:51]: 前面增加一空行，增加可读性

File: src/source_controller/coreservice/core/hostapplyrule/plan.go [hunk 3]
```diff
@@ -48,8 +48,29 @@ func (p *hostApplyRule) GenerateApplyPlan(kit *rest.Kit, bizID int64, option met
...
 	}
+
+	// get attributes
+	attributeIDs := make([]int64, 0)
+	for _, item := range option.Rules {
+		attributeIDs = append(attributeIDs, item.AttributeID)
+	}
+	attributes, err := p.listHostAttributes(kit, bizID, attributeIDs...)
+	if err != nil {
+		blog.ErrorJSON("GenerateApplyPlan failed, listHostAttributes failed, attributeIDs: %s, err: %s, rid: %s",
```
Comment (breezelxp) [inline src/source_controller/coreservice/core/hostapplyrule/plan.go:59]: 不要滥用ErrorJSON，这个地方用errorf就够了

File: src/source_controller/coreservice/core/hostapplyrule/plan.go [hunk 4]
```diff
@@ -48,8 +48,29 @@ func (p *hostApplyRule) GenerateApplyPlan(kit *rest.Kit, bizID int64, option met
...
+		return result, err
+	}
+
+	hostAttrs := []string{}
+	for _, attr :=range attributes{
+		hostAttrs =append(hostAttrs, attr.PropertyName)
+	}
+
+	fields := []string{common.BKHostIDField,common.BKHostInnerIPField,common.BKCloudIDField, common.BKHostNameField}
+	fields = append(fields, hostAttrs...)
```
Comment (breezelxp) [inline src/source_controller/coreservice/core/hostapplyrule/plan.go:70]: 这几行是不是可以合到一块儿去写，不用中间的临时变量？

fields := []string{common.BKHostIDField,common.BKHostInnerIPField,common.BKCloudIDField, common.BKHostNameField}
for _, attr :=range attributes{
		fields =append(fields, attr.PropertyName)
	}

**Review by breezelxp — APPROVED** [review]
LGTM

**Pull request merged**